document the changes in p.a.multilingual regarding translation services and Google Translate. (#2003)
plone/documentation: docs/backend/upgrading/version-specific-migration/upgrade-to-62.md

Reviewer: Oopsie, this file needs has one too many `.` in its name. Please rename so we can see it in the RTD PR preview.

Also it needs to be added to its sibling `index.md` in the `toctree` directive.

---
myst:
  html_meta:
    "description": "How to upgrade to Plone 6.2"
    "property=og:description": "How to upgrade to Plone 6.2"
    "property=og:title": "How to upgrade to Plone 6.2"
    "keywords": "Upgrade, Plone 6"
---

(backend-upgrade-plone-v62-label)=

# Upgrade Plone 6.1 to 6.2

Plone 6.2 has seen the following major changes.
Some may require changes in your setup.

> ## Drop of Google Translate integration

Reviewer: This section should go after the namespaces section that will be added in https://github.com/plone/documentation/pull/2002.

Reviewer: I'll make sure this happens in #2002.

---
myst:
  html_meta:
    "description": "How to upgrade to Plone 6.2"
    "property=og:description": "How to upgrade to Plone 6.2"
    "property=og:title": "How to upgrade to Plone 6.2"
    "keywords": "Upgrade, Plone 6"
---

(backend-upgrade-plone-v62-label)=

# Upgrade Plone 6.1 to 6.2

Plone 6.2 has seen the following major changes.
Some may require changes in your setup.

## Drop of Google Translate integration

`plone.app.multilingual` had an integration to use Google Translate to translate the content in the `babel_view`, where the content in two languages is shown side-by-side.

This integration has been removed in Plone 6.2, and has been exchanged by a generic translation service integration.

> See the {doc}`use-an-external-translation-service-label` chapter for details.

Reviewer: Per the comment from @rohnsha0 and response from @erral, I think this makes it a little more clear.

```suggestion
## Replaced Google Translate integration with generic translation integration

`plone.app.multilingual` had an integration to use only Google Translate to translate the content in the `babel_view`, where the content in two languages is shown side-by-side.
This integration was limited to only Google Translate.
In Plone 6.2, this integration has been replaced with a generic translation service integration hook.

[`collective.translators`](https://github.com/collective/collective.translators) provides some implementations for that hook, supporting AWS, Deepl, Deepseek, Google Translate, Libre Translate, and Ollama.
It can be extended to support more translation providers.

It is not mandatory to use `collective.translator`.
Any developer can write the integration with the tool of their choice.

See the {doc}`use-an-external-translation-service-label` chapter for details.
```

Author: Thanks for rewriting it @stevepiercy, but it needs another round:

```
In Plone 6.2, this integration has been replaced with a generic translation service integration hook.

[`collective.translators`](https://github.com/collective/collective.translators) provides some implementations for that hook, supporting AWS, Deepl, Deepseek, Google Translate, Libre Translate, and Ollama, and can be extended to support more translation providers.

But it is not mandatory to use `collective.translator` any developer can write the integration with the tool of their choice.
```

The idea is not that we have replaced Google Translate with `collective.translators` but with a way where we can hook anything, and collective.translators is an example of it.

Reviewer: Got it, thanks! I tweaked it. Please have another look.

---
myst:
  html_meta:
    "description": "How to upgrade to Plone 6.2"
    "property=og:description": "How to upgrade to Plone 6.2"
    "property=og:title": "How to upgrade to Plone 6.2"
    "keywords": "Upgrade, Plone 6"
---

(backend-upgrade-plone-v62-label)=

# Upgrade Plone 6.1 to 6.2

Plone 6.2 has seen the following major changes.
Some may require changes in your setup.

## Replaced Google Translate integration with generic translation integration

`plone.app.multilingual` had an integration to use only Google Translate to translate the content in the `babel_view`, where the content in two languages is shown side-by-side.
This integration was limited to only Google Translate.
In Plone 6.2, this integration has been replaced with a generic translation service integration hook.

[`collective.translators`](https://github.com/collective/collective.translators) provides some implementations for that hook, supporting AWS, Deepl, Deepseek, Google Translate, Libre Translate, and Ollama.
It can be extended to support more translation providers.

It is not mandatory to use `collective.translator`.
Any developer can write the integration with the tool of their choice.

> See the {doc}`/i18n-l10n/use-an-external-translation-service-label` chapter for details.

Reviewer: C'mon, Steve, get with it!

```suggestion
See the {doc}`/i18n-l10n/use-an-external-translation-service` chapter for details.
```

---
myst:
  html_meta:
    "description": "How to upgrade to Plone 6.2"
    "property=og:description": "How to upgrade to Plone 6.2"
    "property=og:title": "How to upgrade to Plone 6.2"
    "keywords": "Upgrade, Plone 6"
---

(backend-upgrade-plone-v62-label)=

# Upgrade Plone 6.1 to 6.2

Plone 6.2 has seen the following major changes.
Some may require changes in your setup.

## Replaced Google Translate integration with generic translation integration

`plone.app.multilingual` had an integration to use only Google Translate to translate the content in the `babel_view`, where the content in two languages is shown side-by-side.
This integration was limited to only Google Translate.
In Plone 6.2, this integration has been replaced with a generic translation service integration hook.

[`collective.translators`](https://github.com/collective/collective.translators) provides some implementations for that hook, supporting AWS, Deepl, Deepseek, Google Translate, Libre Translate, and Ollama.
It can be extended to support more translation providers.

It is not mandatory to use `collective.translator`.
Any developer can write the integration with the tool of their choice.

> See the {doc}`use-an-external-translation-service-label` chapter for details.

Reviewer: My bad, it's in a different directory.

```suggestion
See the {doc}`../../../i18n-l10n/use-an-external-translation-service-label` chapter for details.
```

> See the {doc}`../../../i18n-l10n/use-an-external-translation-service-label` chapter for details.

Reviewer: Derp, this is simpler.

```suggestion
See the {doc}`/i18n-l10n/use-an-external-translation-service-label` chapter for details.
```

To achieve that integration, the previously existing `gtranslation_service` browser view has been removed from `plone.app.multilingual`.

> Due to not needing it anymore, the `plone.google_translation_key` registry entry has been removed and will be removed by the upgrade step to Plone 6.2.

Reviewer: ```suggestion
Due to not needing it anymore, the `plone.google_translation_key` registry entry has been removed, and it will be removed when performing the upgrade step to Plone 6.2.
```

The `babel_view` has been modified to call a new REST API endpoint instead of the old `gtranslation_service` browser view.

> To keep using the previously existing Google Translate integration, site administrators have to install a new package that provides those and more services, called [`collective.translators`](https://github.com/collective/collective.translators).

Reviewer: I think this sentence is now redundant with the rewritten portion above. @erral do you agree?

```suggestion
```
